src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/RecordingOptionsGetHandler.java

```diff
@@ -0,0 +1,150 @@
+
+import org.openjdk.jmc.common.unit.IConstrainedMap;
+import org.openjdk.jmc.common.unit.IOptionDescriptor;
+import org.openjdk.jmc.flightrecorder.configuration.recording.RecordingOptionsBuilder;
+
+import com.google.gson.Gson;
+
+import com.redhat.rhjmc.containerjfr.commands.internal.RecordingOptionsBuilderFactory;
+import com.redhat.rhjmc.containerjfr.net.AuthManager;
+import com.redhat.rhjmc.containerjfr.net.TargetConnectionManager;
+
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+public class RecordingOptionsGetHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final String PATH = "/api/v1/targets/:targetId/recordingOptions";
+    protected final TargetConnectionManager connectionManager;
+    private final RecordingOptionsBuilderFactory recordingOptionsBuilderFactory;
+    private final Gson gson;
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");
+
+    @Inject
+    RecordingOptionsGetHandler(
+            AuthManager auth,
+            TargetConnectionManager connectionManager,
+            RecordingOptionsBuilderFactory recordingOptionsBuilderFactory,
+            Gson gson) {
+        super(auth);
+        this.connectionManager = connectionManager;
+        this.recordingOptionsBuilderFactory = recordingOptionsBuilderFactory;
+        this.gson = gson;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public String path() {
+        return PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    void handleAuthenticated(RoutingContext ctx) throws Exception {
+        Map<String, String> optionMap =
+                connectionManager.executeConnectedTask(
+                        getConnectionDescriptorFromContext(ctx),
+                        connection -> {
+                            RecordingOptionsBuilder builder =
+                                    recordingOptionsBuilderFactory.create(connection.getService());
+                            IConstrainedMap<String> recordingOptions = builder.build();
+
+                            Map<String, IOptionDescriptor<?>> TargetRecordingOptions =
+                                    connection.getService().getAvailableRecordingOptions();
+
+                            Map<String, String> map = new HashMap<String, String>();
+
+                            if (recordingOptions.get("toDisk") != null) {
+                                map.put("toDisk", recordingOptions.get("toDisk").toString());
+                            } else {
+                                map.put(
+                                        "toDisk",
+                                        TargetRecordingOptions.get("disk").getDefault().toString());
+                            }
+
+                            String maxAge;
+                            if (recordingOptions.get("maxAge") != null) {
```

Comment:
Does the `recordingOptions` have a method like `.containsKey()`? If so, it would be better to use that than `.get() != null`.

Reply:
the IConstrainedMap does not seem to have that method, the only method that I can tell that serves the purpose is .get

Reply:
What about a `.keySet().contains()`?

Reply:
Hmm.

```
	/**
	 * @return A {@link Set set} of keys which are known to be valid. That is, those that currently
	 *         are known to have a {@link IConstraint constraint}. This includes all keys which
	 *         currently have a value, but additional keys may be included. In other words,
	 *         {@link #get(Object)} may return {@code null} for some keys included in this set.
	 */
	Set<K> keySet();
```

Well, nevermind. Just keep using `.get() != null` then.